### ISSUE TITLE
`rgb_to_hsv` returns NaN saturation for black pixels in float16, and `adjust_hue` / `ColorJiggle` / `ColorJitter` spread it

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4560]
`kornia.color.rgb_to_hsv` computes the saturation as `deltac / (max_rgb + eps)` with `eps = 1e-8` (`kornia/color/hsv.py:61`). In `float16` the smallest subnormal is about `6e-8`, so `1e-8` rounds to `0.0`, and a black pixel (`max_rgb == 0`, `deltac == 0`) divides `0 / 0` and gets `NaN` saturation. `float32`, `float64` and `bfloat16` are finite for the same input. Everything built on the HSV round trip inherits it: `kornia.enhance.adjust_hue` on a black pixel is `NaN` in every channel, and `ColorJiggle` / `ColorJitter` with any non-zero `hue` turn every black pixel of a `float16` image into `NaN` — a black border or a masked-out region is enough. The same on `mps`.

## Repro

```python
import torch, kornia.augmentation as K
from kornia.color import rgb_to_hsv
from kornia.enhance import adjust_hue

for dev in ["cpu", "mps"]:
    for dt in [torch.float16, torch.bfloat16, torch.float32]:
        if dev == "mps" and dt is torch.bfloat16:
            continue
        z = torch.zeros(1, 3, 1, 1, device=dev, dtype=dt)
        hsv, hue = rgb_to_hsv(z), adjust_hue(z, 0.1)
        img = torch.rand(1, 3, 6, 8, device=dev, dtype=dt)
        img[..., :2, :3] = 0.0                                   # a black patch
        torch.manual_seed(0)
        out = K.ColorJiggle(0.0, 0.0, 0.0, 0.1, p=1.0)(img)
        if dev == "mps":
            torch.mps.synchronize()
        print(f"{dev} {str(dt):16s} hsv(black)={hsv.flatten().tolist()} adjust_hue(black) nan={hue.isnan().any().item()} "
              f"ColorJiggle nan={int(out.isnan().sum())}/{out.numel()}")
```

## Observed

```
cpu torch.float16    hsv(black)=[0.0, nan, 0.0] adjust_hue(black) nan=True ColorJiggle nan=12/144
cpu torch.bfloat16   hsv(black)=[0.0, 0.0, 0.0] adjust_hue(black) nan=False ColorJiggle nan=0/144
cpu torch.float32    hsv(black)=[0.0, 0.0, 0.0] adjust_hue(black) nan=False ColorJiggle nan=0/144
mps torch.float16    hsv(black)=[0.0, nan, 0.0] adjust_hue(black) nan=True ColorJiggle nan=12/144
mps torch.float32    hsv(black)=[0.0, 0.0, 0.0] adjust_hue(black) nan=False ColorJiggle nan=0/144
```

`torch.finfo(torch.float16)`: `eps = 9.77e-04`, `tiny = 6.10e-05`, smallest subnormal `5.96e-08`; `torch.tensor(1e-8, dtype=torch.float16)` is `0.0`. A grey pixel at `1e-3` is finite (`[0.0, 0.0, 0.0010]`), so the failure is exactly the `max_rgb == 0` case.

## Expected

A black pixel has saturation `0` in every dtype. The guard has to be an argument that is representable in the input's dtype — for example `torch.where(max_rgb == 0, ones, max_rgb)` as the divisor (the pattern the same function already uses for `deltac` at `hsv.py:63`), or an `eps` floored at the dtype's smallest normal — so that the division never sees `0 / 0`. `adjust_hue`, `ColorJiggle` and `ColorJitter` then return the black pixel unchanged, as they do in `float32`.

## Classification

**Broken output** — `NaN` from a valid input. Idiom: 1 (`x + eps` used as a singularity guard, with an `eps` that underflows in the dtype; see #4162 and the `eps` floor pattern).

## Disposition

`n/a — single symbol` — one line in `rgb_to_hsv`; the three callers need no change. Focused fix, welcome as a PR.

## Environment

kornia 0.9.0rc1 @ `90650596` (main), torch 2.14.0, python 3.11, cpu and mps (Apple M1, `PYTORCH_ENABLE_MPS_FALLBACK` unset). Found while writing the 2D intensity convention pins of the augmentation conventions audit (batch 6, tracking issue #4407, sub-batch 6c): the `ColorJiggle` / `ColorJitter` pins skip `float16` on a fixture that contains zeros because of this, and cite this issue as the reason.

Posted on behalf of @ducha-aiki by Claude (Fable 5.1).
